### PR TITLE
{Packaging/MSI} Use embeddable python for MSI

### PR DESCRIPTION
**Pull request by fengzhou-msft**
**Description**  

1. Use [embeddable Python](https://docs.python.org/3/using/windows.html#windows-embeddable) + pip to reduce the size of the bundled Python in MSI.
2. Add a script to test MSI package. The test is not added to CI as it needs Admin Permission to set up(#9270). For now it can be run on a local machine when there's major changes in MSI packaging. 

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Packaging/MSI

File: src/azure-cli/azure/cli/command_modules/ams/tests/latest/test_ams_sp_scenarios.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ class AmsSpTests(ScenarioTest):
     @StorageAccountPreparer(parameter_name='storage_account_for_create')
     @AllowLargeResponse()
     def test_ams_sp_create_reset(self, resource_group, storage_account_for_create):
-        with mock.patch('azure.cli.command_modules.ams._utils._gen_guid', side_effect=self.create_guid):
+        with mock.patch('azure.cli.command_modules.ams.operations.sp._gen_guid', side_effect=self.create_guid):
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/ams/tests/latest/test_ams_sp_scenarios.py:17]: Follow [Where to patch](https://docs.python.org/3/library/unittest.mock.html#where-to-patch)

**Comment by yonzhan** [comment]
Packaging/MSI

File: .gitattributes [hunk 2]
```diff
@@ -13,3 +13,5 @@
 *.bat eol=crlf
 # The az script for Git Bash/Cygwin should be LF
 build_scripts/windows/scripts/az eol=lf
+# sh scripts should be LF
+*.sh eol=lf
```
Comment (haroldrandom) [inline .gitattributes:17]: why need this? Will git applicaiton get conflict with it?

Comment (fengzhou-msft) [reply .gitattributes:17]: I would like reuse the `scripts/ci/build.sh` to run on Git Bash. Without this setting, all files checked out on Windows will be converted to crlf and will have error when run with bash.

**Pull request merged**